src/features/views/components/ViewDataTable/columnTypes/PersonTagColumnType.tsx

```diff
@@ -0,0 +1,93 @@
+import { Box } from '@mui/material';
+import { FC } from 'react';
+import { makeStyles } from '@mui/styles';
+import { useRouter } from 'next/router';
+import { GridColDef, GridRenderCellParams } from '@mui/x-data-grid-pro';
+
+import { IColumnType } from '.';
+import TagChip from 'features/tags/components/TagManager/components/TagChip';
+import TagModel from 'features/tags/models/TagModel';
+import useModel from 'core/useModel';
+import { ZetkinTag } from 'utils/types/zetkin';
+import ZUIFuture from 'zui/ZUIFuture';
+import { PersonTagViewColumn, ZetkinViewRow } from '../../types';
+
+export default class PersonTagColumnType implements IColumnType {
+  cellToString(cell: ZetkinTag): string {
+    return cell.value ? cell.value.toString() : Boolean(cell).toString();
+  }
+
+  getColDef(column: PersonTagViewColumn): Omit<GridColDef, 'field'> {
+    return {
+      align: 'center',
+      headerAlign: 'center',
+      renderCell: (params: GridRenderCellParams<ZetkinTag, ZetkinViewRow>) => {
+        return (
+          <Cell
+            cell={params.value}
+            personId={params.row.id}
+            tagId={column.config.tag_id}
+          />
+        );
+      },
+    };
+  }
+}
+
+const useStyles = makeStyles(() => ({
+  ghost: {
+    pointerEvents: 'none',
+  },
+  ghostContainer: {
+    '&:hover': {
+      opacity: 0.3,
+    },
+    cursor: 'pointer',
+    opacity: 0,
+    transition: 'opacity 0.1s',
+  },
+}));
+
+const Cell: FC<{
+  cell: ZetkinTag | undefined;
+  personId: number;
+  tagId: number;
+}> = ({ cell, personId, tagId }) => {
+  // TODO: Find a way to share a model between cells in a column
```

Comment:
could this maybe be done similarly to how model is passed into to the processRowUpdate funciton?

Reply:
Not very easily, since this is coming from the `renderCell()` function, and we do not control it's signature.